src/pages/EmbeddedWallet.tsx

```diff
@@ -123,36 +138,81 @@ export const EmbeddedWallet = () => {
 
   const isSignupProcessing = isRegistering || isCreatingWallet || isRefreshing;
   const isLoading = isAuthenticating || isSignupProcessing;
+  const inviteToken = token.trim();
+  const hasInviteToken = Boolean(inviteToken);
+
+  const organizationName = useMemo(() => {
+    if (organization?.data?.name) {
+      return organization.data.name;
+    }
+    const tenantName = getSdpTenantName();
+    return tenantName || "Your organization";
+  }, [organization?.data?.name]);
+
+  const organizationInitial = useMemo(() => {
+    const first = organizationName.match(/[A-Za-z0-9]/);
+    return (first ? first[0] : "O").toUpperCase();
+  }, [organizationName]);
+
+  const organizationLogo = organization?.data?.logo;
+
+  const title = hasInviteToken
+    ? "You're invited to create your wallet account"
+    : "Log in to wallet";
+  const subtitle = hasInviteToken
+    ? `Once your wallet is set up, you'll be able to receive funds sent by ${organizationName}. To begin, add a passkey to securely access your account.`
+    : "Use your passkey to log in to wallet";
+
+  const primaryCtaProps = hasInviteToken
+    ? {
+        onClick: handleSignup,
+        disabled: isLoading || !inviteToken,
+        isLoading: isSignupProcessing,
+      }
+    : {
+        onClick: handleLogin,
+        disabled: isLoading,
+        isLoading: isAuthenticating,
+      };
 
   return (
-    <div className="SignIn">
-      <div className="SignIn__container">
-        <div className="SignIn__content">
-          {errorMessage && <Notification variant="error" title={errorMessage} />}
-
-          <Box gap="md">
-            <Button
-              variant="primary"
-              size="lg"
-              onClick={handleSignup}
-              disabled={isLoading || !token.trim()}
-              isLoading={isSignupProcessing}
-            >
-              Create Wallet
-            </Button>
-
-            <Button
-              variant="secondary"
-              size="lg"
-              onClick={handleLogin}
-              disabled={isLoading}
-              isLoading={isAuthenticating}
-            >
-              Sign In
-            </Button>
-          </Box>
+    <EmbeddedWalletLayout
+      organizationName={organizationName}
+      organizationLogo={organizationLogo}
+      headerRight={hasInviteToken ? "Create an account" : undefined}
+      showHeader={hasInviteToken}
+      contentAlign={hasInviteToken ? "left" : "center"}
+    >
+      {errorMessage && <Notification variant="error" title={errorMessage} isFilled />}
+
+      {!hasInviteToken ? (
+        <div className="EmbeddedWalletCard__logo" role="img">
```

Comment:
Role is not needed here. The outer `div` is a container for the `img`, and the initial doesn't need to be an image.

```suggestion
        <div className="EmbeddedWalletCard__logo">
```